build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaBasePlugin.java

```diff
@@ -123,7 +123,7 @@ public static void configureCompile(Project project) {
             // fail on all javac warnings.
             // TODO Discuss moving compileOptions.getCompilerArgs() to use provider api with Gradle team.
             List<String> compilerArgs = compileOptions.getCompilerArgs();
-            compilerArgs.add("-Werror");
+            // compilerArgs.add("-Werror"); NOCOMMIT add me back once we figure out how to not fail compiling with preview features
```

Comment:
This'd be a huge problem to commit, but I can't figure out a good way around it. If I enable the vector API it'll emit the warning. I think Lucene has some kind of hack for accessing the vector API that I think would fix this. And we'd want to steal that.